client/components/domains/register-domain-step/index.jsx

```diff
@@ -392,32 +392,45 @@ class RegisterDomainStep extends React.Component {
 						return callback();
 					}
 
-					checkDomainAvailability( domain, ( error, result ) => {
-						const timeDiff = Date.now() - timestamp;
-						const status = result && result.status ? result.status : error;
-						const { AVAILABLE, UNKNOWN } = domainAvailability;
-						const isDomainAvailable = includes( [ AVAILABLE, UNKNOWN ], status );
+					checkDomainAvailability(
+						{ domainName: domain, blogId: this.props.selectedSite.ID },
```

Comment:
This causes an error during signup (just search for any available domain in signup):
```
Uncaught TypeError: Cannot read property 'ID' of undefined
```

This, of course, also causes the notice for a domain that's being transferred in broken.